src/frame/index.tsx

```diff
@@ -1,28 +1,125 @@
 import { h, Component } from "preact";
-import API from "../api";
 
 type Props = {
-  api: any;
+  apiKey: string;
+  mode: string;
   url: string;
   width?: number;
   height?: number;
+  data?: any;
+  messageKey?: string;
+  messagePayloadType?: string;
+  messagePayloadActionKey?: string;
 };
 
 type State = {};
 
+const encodeURLParams = (url: string, data: any): string => {
+  let params = "";
+  for (const key in data) {
+    if (params !== "") {
+      params += "&";
+    }
+    if (data[key] !== undefined && data[key] !== null) {
+      params += key + "=" + encodeURIComponent(data[key]);
+    }
+  }
+  return `${url}?${params}`;
+};
+
 export default class Frame extends Component<Props, State> {
   static defaultProps = {
     width: "100%",
     height: "100%",
+    mode: "light",
+    data: {},
+    messageKey: "flipside",
+    messagePayloadType: "linkAction",
+    messagePayloadActionKey: "href",
   };
 
+  ref: any = null;
+  setRef = (dom: any) => (this.ref = dom);
+
+  handleMessage = (e: any) => {
+    const eventData = e.data;
+    if (!eventData) return;
+
+    const {
+      messageKey,
+      messagePayloadType,
+      messagePayloadActionKey,
+    } = this.props;
+
+    const message = eventData[messageKey];
+    if (!message) return;
+
+    if (message.type !== messagePayloadType) return;
+
+    const payload = message[messagePayloadType];
+    if (!payload) return;
+
+    const messageAction = payload[messagePayloadActionKey];
+    if (!messageAction || typeof messageAction != "string") return;
+
+    window.location.assign(messageAction);
```

Comment:
We might want this to open in an new tab instead?
```js
window.open(messageAction, '_blank');
```

Reply:
Right now the functionality is to direct the user straight to the link so I wanted to maintain that.